formula/src/main/java/com/instacart/formula/internal/CallbackImpl.kt

```diff
@@ -0,0 +1,37 @@
+package com.instacart.formula.internal
+
+import com.instacart.formula.Listener
+import com.instacart.formula.Update
+import java.lang.IllegalStateException
+
+/**
+ * Note: this class is not a data class because equality is based on instance and not [key].
+ */
+@PublishedApi
+internal class CallbackImpl<State, Event>(internal var key: Any) : Listener<Event> {
```

Comment:
TODO: this should be renamed to `ListenerImpl` or `Listener` should be renamed to `Callback`

Reply:
Maybe add this as a comment to this class?

Reply:
Most likely, I'm going to rename this before merging the PR.